### ISSUE TITLE
Should batch change names be case-insensitive unique?

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#45646]
We've had a TODO comment about that question in the code base for long and honestly, I think it should be. What do others think? Might be nasty to migrate though.

**Comment by courier-new** [comment]
I agree, I'd expect `my-batch-change` === `My-Batch-Change`, like usernames or emails that differ only in case are not unique. But this is also probably pretty low priority, especially if it's non-trivial... so... backlog? 😅

**Comment by eseliger** [comment]
it should be easy from an implementation perspective, we make the name column citext instead and then add some nice err handling to it and done. The nasty part would be what to do about batch changes that have a name clash if we would switch to case insensitivity 😬 maybe not too bad, idk. We could just rename those cases to -2 or so, assuming literally no one did it anyways 😆 All over all, I guess this would be 1.5d maybe - but no strong preference about it's priority.

**Comment by courier-new** [comment]
We've agreed for handling name conflicts we want to rename *both* items, like `batch-change-1` and `batch-change-2`, rather than letting one of them "win" the original name. This means existing links will break, rather than point to the wrong resource.
